Detectors/TPC/workflow/src/ClustererSpec.cxx

```diff
@@ -218,7 +239,16 @@ DataProcessorSpec getClustererSpec(bool sendMC, bool haveDigTriggers)
   return DataProcessorSpec{processorName,
                            {createInputSpecs(sendMC, haveDigTriggers)},
                            {createOutputSpecs(sendMC)},
-                           AlgorithmSpec(initFunction)};
+                           AlgorithmSpec(initFunction),
+                           Options{
+                             {"peakChargeThreshold", VariantType::Int, 2, {"Charge threshold for the central peak in ADC counts"}},
+                             {"contributionChargeThreshold", VariantType::Int, 0, {"Charge threshold for the contributing pads in ADC counts"}},
+                             {"splittingMode", VariantType::Int, 0, {"cluster splitting mode, 0 no splitting, 1 for minimum contributes half to both, 2 for miminum corresponds to left/older cluster"}},
+                             {"isContinuousReadout", VariantType::Bool, true, {"Switch for continuous readout"}},
+                             {"rejectSinglePadClusters", VariantType::Bool, false, {"Switch to reject single pad clusters, sigmaPad2Pre == 0"}},
+                             {"rejectSingleTimeClusters", VariantType::Bool, false, {"Switch to reject single time clusters, sigmaTime2Pre == 0"}},
+                             {"rejectLaterTimebin", VariantType::Bool, false, {"Switch to reject peaks in later timebins of the same pad"}},
+                           }};
 }
```

Comment:
Do we really need to define options in this way? Note that in many places we already use a self-documenting [ConfigurableParam](https://github.com/AliceO2Group/AliceO2/blob/dev/Common/SimConfig/include/SimConfig/ConfigurableParam.h)-based objects (eventually @sawenzel plans to interface it to ccdb).
See for example how the [interaction diamond](https://github.com/AliceO2Group/AliceO2/blob/dev/Generators/include/Generators/InteractionDiamondParam.h) or [ITS Alpide params](https://github.com/AliceO2Group/AliceO2/blob/dev/Generators/include/Generators/InteractionDiamondParam.h) are used to provide settings for [PrimaryGenerator](
https://github.com/AliceO2Group/AliceO2/blob/dev/Generators/src/PrimaryGenerator.cxx#L64) or [ITS digitizations device](https://github.com/AliceO2Group/AliceO2/blob/Steer/DigitizerWorkflow/src/ITSMFTDigitizerSpec.cxx#L300). To trigger the update of parameters from the command line or ini file you just need to issue in the workflow [definition](https://github.com/AliceO2Group/AliceO2/blob/c6b4a1257fa465b5966fdb58221c14c46ec9cb41/Steer/DigitizerWorkflow/src/SimpleDigitizerWorkflow.cxx#L305):
````
o2::conf::ConfigurableParam::updateFromFile(configcontext.options().get<std::string>("configFile"));
o2::conf::ConfigurableParam::updateFromString(configcontext.options().get<std::string>("configKeyValues"));
o2::conf::ConfigurableParam::writeINI("o2digitizerworkflow_configuration.ini");
````
with ``configKeyValues`` and ``configFile`` options defined once (for all types of params) in the [workflow](https://github.com/AliceO2Group/AliceO2/blob/Steer/DigitizerWorkflow/src/SimpleDigitizerWorkflow.cxx#L141)

Reply:
Hi Ruben, indeed, very nice Idea! I implemented it!